src/test/shell/bazel/starlark_repository_test.sh

```diff
@@ -2854,4 +2854,61 @@ EOF
   expect_log "Circular definition of repositories"
 }
 
+function test_watch_file_status_change() {
```

Comment:
Mind adding a test case that tests what happens when you watch a file whose containing directory does not exist and one where you watch a file that has a dangling symlink during symlink resolution?

I.e. `rctx.watch("a/b/c")`, then

```
rm -fr a
mkdir a
bazel build ...

mkdir a/b
touch a/b/c
bazel build ...

rm -fr a/b
ln -s d a/b
bazel build ...

mkdir d
touch d/c
bazel build ...
```
As far as I can tell, it works because `FileValue` gets invalidated appropriately and when there is a server restart, the marker file is re-read, but let's test this particular case.

It'll get more interesting when you'll also start checking mtimes because then you'll probably have to check inodes for this to be correct in not-so-unusual cases (e.g `echo a > a; echo b >b; ln -s a watched; bazel build; ln -s b watched`), but for now, the digest-based checking covers all the bases.

Reply:
...also add a test case for dangling symlinks, i.e. `ln -s nonexistent a/b/c; bazel build ...; touch nonexistent; bazel build ...`

Reply:
thanks for the great suggestions! added (almost verbatim...)

Reply:
I was inclined to also ask for some testing of behavior over server restarts, but I'll stop annoying you :)